### PR TITLE
Use Setenv as well

### PR DESCRIPTION
**Pull request by Crell**
See #9 for background.

### CODE REVIEW

**Comment by pjcdawkins** [comment]
Is `$_SERVER` otherwise a normal way to access environment variables? not `$_ENV`?

**Comment by Crell** [comment]
Unfortunately PHP is rather inconsistent. :frowning: 

At startup, it populates $_ENV, $_SERVER, and getenv(), UNLESS it's running through the built in web server in which case only getenv() works.  (Or maybe $_SERVER does as well, I forget.)

After startup they are not kept in sync at all, and all can be changed independently.

I don't know what constitutes "normal way to access"; I'd say getenv(), but I cannot speak for what every system is doing.

**Comment by pjcdawkins** [comment]
> At startup, it populates $_ENV, $_SERVER, and getenv(), UNLESS it's running through the built in web server in which case only getenv() works. (Or maybe $_SERVER does as well, I forget.)

FTR it depends on the `variables_order` ini setting, so the built-in server populates those variables if you use

```sh
php -d variables_order=egpcs -S 127.0.01:8000 -t docroot
```

So... if I was being picky I'd suggest something like:

```php
function addToEnv(string $name, string $value): void {
    if (!putenv("$name=$value")) {
        throw new \RuntimeException('Failed to create environment variable: '.$name);
    }
    $order = ini_get('variables_order');
    if (stripos($order, 'e') !== false) {
        $_ENV[$name] = $value;
    }
    if (stripos($order, 's') !== false) {
        if (strpos($name, 'HTTP_') !== false) {
            throw new \RuntimeException('Refusing to add ambiguous environment variable '.$name.' to $_SERVER');
        }
        $_SERVER[$name] = $value;
    }
}
```

But I notice Symfony doesn't care:
https://github.com/symfony/symfony/blob/b1a41a4812663910912f11a0e9fe10e0c9f96ef6/src/Symfony/Component/HttpKernel/Kernel.php#L113-L117

**Comment by Crell** [comment]
I don't mind picky. :smile: Let's do that.  Tests still pass.

File: platformsh-flex-env.php [hunk 1]
```diff
@@ -18,21 +18,49 @@ function mapPlatformShEnvironment() : void
...
-        $_SERVER['APP_SECRET'] = getenv('PLATFORM_PROJECT_ENTROPY');
-    }
+    $secret = getenv('APP_SECRET') ?: getenv('PLATFORM_PROJECT_ENTROPY') ?: null;
+    setEnvVar('APP_SECRET', $secret);
 
     // Default to production. You can override this value by setting
     // `env:APP_ENV` as a project variable, or by adding it to the
     // .platform.app.yaml variables block.
-    $_SERVER['APP_ENV'] = $_SERVER['APP_ENV'] ?? (getenv('APP_ENV') ?: null) ?? 'prod';
+    $appEnv = $_SERVER['APP_ENV'] ?? getenv('APP_ENV') ?: 'prod';
```
Comment (pjcdawkins) [inline platformsh-flex-env.php]:
```php
$appEnv = getenv('APP_ENV') ?: 'prod';
```

File: platformsh-flex-env.php [hunk 2]
```diff
@@ -18,21 +18,49 @@ function mapPlatformShEnvironment() : void
...
+    setEnvVar('APP_ENV', $appEnv);
 
     if (!isset($_SERVER['DATABASE_URL'])) {
```
Comment (pjcdawkins) [inline platformsh-flex-env.php]:
```php
if (!getenv('DATABASE_URL')) {
```

**Pull request merged**